cmp/grouping/GroupingChooserModel.ts

```diff
@@ -183,10 +183,21 @@ export class GroupingChooserModel extends HoistModel {
             }
         });
 
+        this.addReaction({
+            track: () => this.dimensions,
+            run: () => this.cleanStaleDims()
+        });
+
         this.setValue(value);
         this.setFavorites(favorites);
     }
 
+    @action
+    setDimensions(dimensions: Array<DimensionSpec | string>) {
+        this.dimensions = this.normalizeDimensions(dimensions);
+        this.dimensionNames = keys(this.dimensions);
+    }
```

Comment:
We should use this in the ctor vs. duplicating code

Reply:
It would be safer to use `_dimensions` as a private property to store the dims, with this setter and a public `get dimensions()` getter, to ensure this method is called when changing - i.e. they are not changed directly.  

Then could remove the new reaction and call `cleanStaleDims` here - more direct / synchronous